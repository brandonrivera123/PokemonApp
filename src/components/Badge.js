import React from "react";
import { colors } from "theme";
import styles from "./styles/badge.module.css";

export const Badge = ({ item }) => {
  return (
    <div
      className={styles.type}
      style={{
        color: `${colors[item.type.name] ? colors[item.type.name] : "white"}`,
      }}
    >
      {item.type.name}
    </div>
  );
};
