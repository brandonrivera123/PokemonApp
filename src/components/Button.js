import React from "react";
import { Ripple } from ".";
import styles from "./styles/button.module.css";

export const Button = ({ name = "button", ...props }) => {
  return (
    <button className={styles.button} {...props}>
      <Ripple />
      <span data-testid="button-name" className={styles.button__span}>
        {name}
      </span>
    </button>
  );
};
