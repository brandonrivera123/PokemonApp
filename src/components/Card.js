import React from "react";
import { Loader, Badge } from ".";
import { useNavigate } from "react-router-dom";
import types from "styles/type.module.css";
import styles from "./styles/card.module.css";

export const Card = ({
  img,
  name,
  id,
  // description,
  loading,
  type,
}) => {
  const navigate = useNavigate();

  return (
    <div
      className={styles.card}
      onClick={() => navigate(`stats/${id}/${name}`)}
    >
      {!loading ? (
        <div
          className={styles.card__media}
          style={{ backgroundImage: `url(${img})` }}
        />
      ) : (
        <Loader />
      )}
      <div className={`${styles.card__content} ${types[type[0].type.name]}`}>
        {!loading ? (
          <>
            <h2 data-testid="card-name" className={styles.pokemonName}>
              {name}
            </h2>
            <div className={styles["type-container"]}>
              {type.map((item, index) => (
                <Badge key={index} item={item} />
              ))}
            </div>
          </>
        ) : (
          <h2 className={styles.pokemonName}>...Loading</h2>
        )}
        {/* <hr /> */}
        {/* {!loading ? (
          <p style={{ fontSize: "1.25rem" }}>{description}</p>
        ) : (
          <p tyle={{ fontSize: "1.25rem" }}>...Loading</p>
        )} */}
      </div>
    </div>
  );
};
