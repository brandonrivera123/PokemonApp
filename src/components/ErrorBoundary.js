import React from "react";
import styles from "styles/error.module.css";

class ErrorBoundary extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = { hasError: false, error: null, errorInfo: null };
  }

  static getDerivedStateFromError(error) {
    // Update state so the next render will show the fallback UI.
    return { hasError: true };
  }

  componentDidCatch(error, errorInfo) {
    // You can also log the error to an error reporting service
    //   logErrorToMyService(error, errorInfo);
    this.setState({
      error: error,
      errorInfo: errorInfo,
    });
  }

  render() {
    if (this.state.hasError) {
      // You can render any custom fallback UI
      return (
        <div className={styles.container}>
          <div>
            <h1 style={{ color: "white", textAlign: "center" }}>
              Something went wrong.
            </h1>
            <div className={styles.error}>
              {this.state.error && this.state.error.toString()}
            </div>
            <div className={styles.error__info}>
              {this.state.errorInfo && this.state.errorInfo.componentStack}
            </div>
          </div>
        </div>
      );
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
