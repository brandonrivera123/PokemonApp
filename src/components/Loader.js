import React from "react";
import styles from "./styles/loader.module.css";

export const Loader = () => {
  return (
    <div
      className={styles.loader}
      style={{ background: "#333333", height: "140px" }}
    >
      <div className={styles["lds-ripple"]}>
        <div></div>
        <div></div>
      </div>
    </div>
  );
};
