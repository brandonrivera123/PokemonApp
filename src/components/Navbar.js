import React from "react";
import { useLocation } from "react-router-dom";
import pokeImage from "../assets/pokeapi.png";
import styles from "./styles/navbar.module.css";

export const Navbar = () => {
  const location = useLocation();
  return (
    <div className={styles.nav}>
      <img className={styles.nav__title} src={pokeImage} alt="poketitle" />
      {location.pathname === "/" && (
        <input className={styles.nav__input} type="text" />
      )}
    </div>
  );
};
