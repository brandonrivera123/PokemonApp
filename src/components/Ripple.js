import React, { useState, useLayoutEffect } from "react";
import PropTypes from "prop-types";
import styles from "./styles/ripple.module.css";

/**
 * Adds Ripples to component
 * @param {number} rippleCount // Number of ripples
 * @param {number} duration // Duration of ripple effect
 * @param {function} cleanUpFunction // Function to clean up ripple effects
 */
const useDebouncedRippleCleanUp = (rippleCount, duration, cleanUpFunction) => {
  useLayoutEffect(() => {
    let bounce = null;
    if (rippleCount > 0) {
      clearTimeout(bounce);

      bounce = setTimeout(() => {
        cleanUpFunction();
        clearTimeout(bounce);
      }, duration * 4);
    }

    return () => clearTimeout(bounce);
  }, [rippleCount, duration, cleanUpFunction]);
};

/**
 *
 * @param {*} param0
 * @returns
 */
export const Ripple = ({ duration = 850, color = "#fff", ...props }) => {
  const [rippleArray, setRippleArray] = useState([]);

  useDebouncedRippleCleanUp(rippleArray.length, duration, () => {
    setRippleArray([]);
  });

  const addRipple = (event) => {
    // const testdocument = document.body.getBoundingClientRect();
    // console.log(event.pageX);
    // console.log(event.pageY);
    // console.log(window.scrollY);
    const rippleContainer = event.currentTarget.getBoundingClientRect();
    // console.log(rippleContainer);
    const size =
      rippleContainer.width > rippleContainer.height
        ? rippleContainer.width
        : rippleContainer.height;
    const diameter = size / 2;
    const x = event.pageX - rippleContainer.x - diameter;
    const y = event.pageY - rippleContainer.y - window.scrollY - diameter;
    const newRipple = {
      x,
      y,
      size,
    };

    setRippleArray([...rippleArray, newRipple]);
  };

  return (
    <div
      className={styles.container}
      duration={duration}
      color={color}
      onMouseDown={addRipple}
      {...props}
    >
      {rippleArray.length > 0 &&
        rippleArray.map((ripple, index) => {
          return (
            <span
              key={"span" + index}
              style={{
                top: ripple.y,
                left: ripple.x,
                width: ripple.size,
                height: ripple.size,
              }}
            />
          );
        })}
    </div>
  );
};

Ripple.propTypes = {
  duration: PropTypes.number,
  color: PropTypes.string,
};
