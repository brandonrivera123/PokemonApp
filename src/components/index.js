export * from "./Ripple";
export * from "./Button";
export * from "./Card";
export * from "./Loader";
export * from "./Tooltip";
export * from "./Navbar";
export * from "./Badge";
