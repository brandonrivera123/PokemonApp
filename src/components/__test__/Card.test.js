import { render, screen, waitFor } from "@testing-library/react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import { Card } from "components";

describe("Smoke Tests", () => {
  it("renders card component", async () => {
    await waitFor(() => {
      render(
        <BrowserRouter>
          <Card name="test" type={[{ type: { name: "poison" } }]} />
        </BrowserRouter>
      );
    });
    const nameElement = screen.getByTestId("card-name");
    expect(nameElement).toBeInTheDocument();
    expect(nameElement).toHaveTextContent(/^test$/);
  });
  it("matches snapshot", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Card type={[{ type: { name: "poison" } }]} />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
