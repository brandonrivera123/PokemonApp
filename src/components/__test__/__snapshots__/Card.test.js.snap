// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke Tests matches snapshot 1`] = `
<div
  className="card"
  onClick={[Function]}
>
  <div
    className="card__media"
    style={
      Object {
        "backgroundImage": "url(undefined)",
      }
    }
  />
  <div
    className="card__content poison"
  >
    <h2
      className="pokemonName"
      data-testid="card-name"
    />
    <div
      className="type-container"
    >
      <div
        className="type"
        style={
          Object {
            "color": "purple",
          }
        }
      >
        poison
      </div>
    </div>
  </div>
</div>
`;
